### DISCUSSION TITLE
My own Local Energy Management System

### DISCUSSION THREAD

**Discussion by xavi104** [wills106/homeassistant-solax-modbus#276] [Q&A]
Hi all,

I am implemating my own EMS (energy management system) using a modbus local controller in HA. I was working well with my old Solax G3 X1 using the registers 0x0051 to enable/disable and the 0x0052 to set the active power target. 
Now I am working with a new Solax G4 X1 and seems that the method has changed, anyone has the new modbus map? do I have to implement any special procedure to active the local modbus control mode?

thanks in advance!

**Comment by wills106** [comment]
Have a look at the [Wiki](https://github.com/wills106/homeassistant-solax-modbus/wiki/Modbus-Power-Control-(remotecontrol-power))

**Reply by xavi104** [reply]
Thanks wills106, I saw, at the wiki, that with your custom component you have implemented just this. Because we are trying to learn about how I can control the Solax, I would like to understand how I can transmit the active power target to the inverter through a simple automation that I could use in HA using the modbus.write_register services. I don't know how to write as a WRITE_DATA_LOCAL. I have been looking into your custom componet files but I don't have enough knowledge to figure out the method to send this data to the inverter. Appreciate your support if it is possible.

thanks

**Reply by wills106** [reply]
I have never been given Modbus documents from SolaX.

But between the Wiki and plugin_solax you should be able to workout what's going on.

I gave up using the built in HA Modbus along time ago, I got fedup of breaking changes and didn't agree with some of the other changes that were made.
I don't know how much support there is for writing multiple registers in a single transaction.

**Reply by xavi104** [reply]
Thanks so much wills106, I will explore with this information.
